core/src/Pos/Core/Update/Vote.hs

```diff
@@ -115,4 +116,7 @@ checkUpdateVote pm it =
   where
     sigValid = checkSig pm SignUSVote (uvKey it) (uvProposalId it, uvDecision it) (uvSignature it)
 
+mkVoteId :: UpdateVote -> VoteId
```

Comment:
I moved this over from `Pos.Core.Update.VoteId` but I'm unsure that this is where it should be. The actual type definition for `VoteId` is now in `Pos.Core.Update.Proposal` for [this reason](#r206333757).

I wonder if all of the content of this module should just be in `Pos.Core.Update.Proposal`.

Reply:
Forgot to mention that `mkVoteId` cannot be moved into `Pos.Core.Update.Proposal` along with the definition of `VoteId` because it depends on `UpdateVote` which is defined in this module (`Pos.Core.Update.Vote`) and this module already depends on `Pos.Core.Update.Proposal` (circular dependency).

Reply:
Yeah, that sounds like a bit of a pickle. Moving it here seems fine.

Reply:
It does seem like we could just move `Pos.Core.Update.Payload` into `Pos.Core.Update.Vote` and reduce the number of modules, since they are very tightly couple anyways. `Pos.Core.Update.Payload` is only imported in 3 places, anyways

```
core/src/Pos/Core/Update/Vote.hs
core/src/Pos/Core/Update/Payload.hs
core/src/Pos/Core/Update.hs
```

(1) is obvious. (2) imports both `VoteId` and `Payload`, so if we combine them, no problem. (3) just re-exports the module, and no-one else imports it directly, so no-one else would notice if we combine.

Combining seems fine, if it eliminates some pain. It seems like the coupled-but-separate modules have cause @intricate some difficulties already, so we can just cut that off at the root.

Reply:
I've gone ahead and combined `Proposal` and `Vote`.